rfcs/0046-platform-support-tiers.md

```diff
@@ -0,0 +1,272 @@
+
+Platform-specific fixes are expected to be rare and non-intrusive. Updates
+might break builds.
+
+#### Tier 3
+
+Fixes necessary for this platforms must be either limited to compilation
+toolchains, or general cleanups of non-standard assumptions (e.g. «everything
+that is no x86 is a kind of ARM» or «malloc(0) behaviour is a reliable
+indicator of other malloc features»). These fixes must be generic: there
+should be a reasonable expectation that other exotic platforms would equally
+benefit from the exact same fix.
+
+#### Tier 4
+
+It is recommended to keep platform-specific patches to the toolchain in a
+separate package. Cleanups not necessary on any tier-3 platforms can be
+rejected.
+
+### The number of working packages
+
+#### Tier 1
+
+Almost everything that is not explicitly platform specific and that is not
+abandoned (in Nixpkgs) works.
+
+#### Tier 1.5
+
+Most packages work, credible ambition to reach Tier 1 at some point.
+
+#### Tier 2
+
+Most of the popular packages work.
+
+#### Tier 3
+
+Some packages work.
+
+#### Tier 4
+
+Platform definitions present, a small number of packages might be working.
+
+## Current platforms
+
+### Tier 1
+
+Developer/user base: most of the Nix developers/users
+
+* `x86_64-linux`, `gcc`+`glibc`
+
+### Tier 1.5
+
+Fewer developers and users, less testing; tier-1 tooling.
+
+* `aarch64-linux`, `gcc`+`glibc`
+
+A team @NixOS/aarch64-maintainers shall be created to include people who
+understand the platform and use it.
+
+If there is a complicated problem on this platform when updating a package
+that was previously built succesfully on Aarch64, @NixOS/aarch64-maintainers
+team should be informed.
+
+* `x86_64-darwin`, `clang`+Darwin/macOS
+
+If there is a complicated problem on this platform when updating a package
+that was previously built succesfully on macOS, @NixOS/darwin-maintainers team
+should be informed.
+
+### Tier 2-ε
+
+Pure stdenv for native builds is a channel-blocking job; `wine` dependencies
+are available in the binary cache
+
+* `i686-linux`, `gcc`+`glibc` — `ofBorg` builds via `pkgsi686Linux`, binary
+  cache contains `wine` dependencies
+
+### Tier 2
+
+* `armv{6,7,8}*-linux`, `gcc`+`glibc`
+
+* `mipsel-linux`, `gcc`+`glibc`
+
+* `x86_64-linux`, `gcc`+`musl`
+
+### Tier 3
+
+* `aarch64-none`
+
+* `avr`
+
+* `arm-none`
+
+* `i686-none`
+
+* `x86_64-none`
+
+* `powerpc-none`
+
+* `powerpcle-none`
+
+* `x86_64-mingw32`
+
+* `i686-mingw32`
+
+* `x86_64-linux`, `gcc`+`musl` — static
+
+* `x86_64-linux`, `clang`+`glibc`
+
+* `x86_64-linux`, `clang`+`glibc` — `llvm` linker
+
+* `x86_64-linux` — Android
+
+* `aarch64-linux` — Android
+
+* `armv{7,8}-linux` — Android
+
+### Tier 4
+
+* `x86_64-linux`, `gcc`+`glibc` — static
+
+* `x86_64-linux`, `gcc`+`glibc` — `llvm` linker
+
+### Tier 4.5
+
+Work ongoing to provide/merge Tier 4 support
+
+* `wasm-wasi`
+
+* `powerpc64le-linux`, `gcc`+`glibc`
+
+### Tier 5
+
+No current support, but previous support or clear path to add support
```

Comment:
Illumos / Solaris (we’re inconsistent about what we call it in Nixpkgs) should be added here.

Reply:
What is the platform identifier you would pick here? It is not really a commitment to anything anyway.

Reply:
Let’s go with `x86_64-illumos` and `i686-solaris`, because I think that’s what we actually have targeted in the past. They should be differentiated because they’ve significantly diverged at this point.